src/main/groovy/com/ullink/NuGetSpec.groovy

```diff
@@ -65,24 +61,44 @@ class NuGetSpec extends Exec {
     }
 
     String supplementDefaultValueOnNuspec(String nuspecString) {
+        def final msbuildTaskExists = project.tasks.findByName('msbuild') != null
 
         def root = new XmlSlurper(false, false).parseText(nuspecString)
 
         def defaultValues = {}
-        def applyDefaultValue = { String node, String value ->
+        def setDefaultMetadata = { String node, String value ->
             if (root.metadata[node].isEmpty()) {
                 defaultValues <<= { delegate."$node" value }
             }
         }
-        applyDefaultValue ('id', project.name)
-        applyDefaultValue ('version', project.version)
-        applyDefaultValue ('description', project.description)
-
-        if (root.metadata.isEmpty()) {
-            root.appendNode { metadata defaultValues }
-        } else {
-            root.metadata.appendNode defaultValues
+
+        setDefaultMetadata ('id', project.name)
+        setDefaultMetadata ('title', project.name)
+        setDefaultMetadata ('version', project.version)
+        setDefaultMetadata ('description', project.description ? project.description : project.name)
+
+        def appendAndCreateParentIfNeeded = {
+            parentNodeName, children ->
+                if (root."$parentNodeName".isEmpty()) {
+                    root.appendNode { "$parentNodeName" children }
+                } else {
+                    root."$parentNodeName".appendNode children
+                }
         }
+
+        appendAndCreateParentIfNeeded('metadata', defaultValues)
+
+        if (msbuildTaskExists) {
+            def defaultFiles = {}
+            project.msbuild.mainProject.dotnetArtifacts.each {
+                artifact ->
+                    def fwkFolderVersion = project.msbuild.mainProject.properties.TargetFrameworkVersion.toString().replace('v', '').replace('.', '')
+                    defaultFiles <<= { file(src: artifact.toString(), target: 'lib/net' + fwkFolderVersion) }
+            }
+
+            appendAndCreateParentIfNeeded('files', defaultFiles)
```

Comment:
I missed the fact that this provide no way to choose which files gets added (all output from msbuild is added, always, and you can't exclude it), you can notice the effect of ILRepack build:
```  <files>
    <file src="C:\workspace\git-perso\il-repack\build\tmp\repack\ILRepack.dll" target="lib/net40"/>
    <file src="C:\workspace\git-perso\il-repack\ILRepack\bin\Release\ILRepack.exe" target="lib/net40"/>
    <file src="C:\workspace\git-perso\il-repack\ILRepack\bin\Release\ILRepack.pdb" target="lib/net40"/>
  </files>```